### ISSUE TITLE
Enable a sphinx-autodoc build

### ISSUE THREAD

**Issue by bernhardreiter** [certtools/intelmq#910]
Now that we have #857 decided we should add instructions or configuration to build the documentation from the docstrings. 

Today I've started to look at it a bit and learned:
 * First, we need a sphinx version >= 1.3b1 ([released Oct 10, 2014](http://www.sphinx-doc.org/en/stable/changes.html)) for sphinx.ext.napoleon to be included.
 * Both Debian Jessie and Ubuntu 14.04LTS Trusty versions are too old, thus example instructions should assume use of a venv.
 * When generating autodoc documentation the modules get imported, thus all imports of the modules to be documented must be available, too.

**Comment by bernhardreiter** [comment]
https://pypi.python.org/pypi/travis-sphinx/1.4.3 may come in handy at some point.

**Comment by bernhardreiter** [comment]
The dirty recipe outlined in https://github.com/Intevation/intelmq-mailgen/issues/38 als works.

 * Make sure you have an uptodate sphinx-doc (tested with current 1.5.3).
 * The intelmq dependencies have to be installed so all modules can be imported that you want to generate an apidoc for. (Hint you can look at `.travis.yml` and the pip install lines in there.)
 * Run a dirty command, e.g. like

```sh
sphinx-apidoc --private --full --append-syspath . --output-dir apidocs \
  --implicit-namespaces . setup.py
cd apidocs
rm .rst  # in my tests the contents was not useful
vim conf.py # to add sphinx.ext.napoleon
make html
pushd _build/html/ ; python3 -m http.server 8018 ; popd
```
on simple changes `make html` is enough to rebuild the changed files.

**Comment by wagner-certat** [comment]
Thanks for your work! Works fine for me too.

> ```sh
> rm .rst
> ```
I do not have such a file

**Comment by bernhardreiter** [comment]
@wagner-certat 
about `.rst`: this file is not created, if `setup.py` is used in the command, so in the intermediate version of the instructions had an superflouos `rm`. Should be fine now.

**Comment by bernhardreiter** [comment]
@wagner-certat I'm just preparing a pull-request to eliminate the warnings in master.

**Comment by wagner-certat** [comment]
Copying the instructions to the devguide is quite easy.
My question is: Do we want to provide a built documentation somewhere? I can only push them to some branch on github.
Automatic builds on travis are also possible, except for the manual `conf.py`-changes it should be easy.

@aaronkaplan Are there plans/ideas to put this on the website?

**Comment by aaronkaplan** [comment]
looks good, but website will be re-visited in the summer at the earliest. -> moving to 1.1

**Comment by aaronkaplan** [comment]
Should we put that into 3.0? Looks good and actually looks ready.

**Comment by wagner-certat** [comment]
fixed by #1622
